katharsis-client/src/main/java/io/katharsis/client/internal/DefaultQueryParamsSerializer.java

```diff
@@ -0,0 +1,119 @@
+package io.katharsis.client.internal;
+
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Map.Entry;
+import java.util.Set;
+
+import io.katharsis.queryParams.QueryParams;
+import io.katharsis.queryParams.RestrictedPaginationKeys;
+import io.katharsis.queryParams.RestrictedSortingValues;
+import io.katharsis.queryParams.include.Inclusion;
+import io.katharsis.queryParams.params.FilterParams;
+import io.katharsis.queryParams.params.IncludedFieldsParams;
+import io.katharsis.queryParams.params.IncludedRelationsParams;
+import io.katharsis.queryParams.params.SortingParams;
+import io.katharsis.queryParams.params.TypedParams;
+import io.katharsis.resource.RestrictedQueryParamsMembers;
+
+public class DefaultQueryParamsSerializer implements QueryParamsSerializer {
+
+	@Override
+	public Map<String, Set<String>> serializeFilters(QueryParams queryParams) {
+		Map<String, Set<String>> map = new HashMap<String, Set<String>>();
+		TypedParams<FilterParams> filters = queryParams.getFilters();
+		if (filters != null) {
+			Map<String, FilterParams> filterParamsMap = filters.getParams();
+			for (Map.Entry<String, FilterParams> entry : filterParamsMap.entrySet()) {
+				String type = entry.getKey();
+				Map<String, Set<String>> params = entry.getValue().getParams();
+				for (Map.Entry<String, Set<String>> param : params.entrySet()) {
+					String paramName = RestrictedQueryParamsMembers.filter.name() + "[" + type + "]" + serializeKey(param.getKey());
+					map.put(paramName, param.getValue());
+				}
+			}
+		}
+		return map;
+	}
+
+	private String serializeKey(String key) {
+		return "[" + key.replace(".", "][") + "]";
+	}
+
+	@Override
+	public Map<String, String> serializeSorting(QueryParams queryParams) {
+		Map<String, String> map = new HashMap<String, String>();
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Replace the type specification in this constructor call with the diamond operator ("<>"). [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS2293)